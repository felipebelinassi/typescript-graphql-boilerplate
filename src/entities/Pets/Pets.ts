import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity({ name: 'pets' })
export default class Pets {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ type: 'text' })
  name!: string;

  @Column({ type: 'text', nullable: true })
  breed?: string;

  @Column({ type: 'text' })
  species!: string;

  @Column({ type: 'date', name: 'birth_date' })
  birthDate!: string;
}
