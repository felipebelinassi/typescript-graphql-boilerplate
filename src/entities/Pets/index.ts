import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity({ name: 'pets' })
export default class Pets {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ type: 'text' })
  name!: string;

  @Column({ type: 'text', nullable: true })
  breed?: string;

  @Column({ type: 'text' })
  species!: string;

  @Column({ type: 'date', name: 'birth_date' })
  birthDate!: string;

  @CreateDateColumn({
    type: 'timestamptz',
    name: 'created_at',
    default: () => 'CURRENT_TIMESTAMP',
  })
  createdAt!: Date;
}
