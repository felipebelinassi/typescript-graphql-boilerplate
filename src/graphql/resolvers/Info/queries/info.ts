import { Resolver, Query } from 'type-graphql';
import AppInfo from '../types/objects/Info';

@Resolver()
class InfoResolver {
  @Query(() => AppInfo, { description: 'API info' })
  info(): AppInfo {
    return {
      port: Number(process.env.PORT),
      healthy: true,
    };
  }
}

export default InfoResolver;
