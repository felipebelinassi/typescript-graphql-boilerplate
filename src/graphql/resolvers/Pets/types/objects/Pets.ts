import { ObjectType, Field, Int } from 'type-graphql';

@ObjectType()
export default class Pets {
  @Field(() => Int)
  id!: number;

  @Field()
  name!: string;

  @Field({ nullable: true })
  breed?: string;

  @Field({ nullable: true })
  species!: string;

  @Field()
  birthDate!: string;

  @Field()
  createdAt!: Date;
}
