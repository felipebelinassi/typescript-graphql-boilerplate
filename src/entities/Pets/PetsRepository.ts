import { EntityRepository, Repository } from 'typeorm';
import Pets from '.';
import CreatePetParams from '../../graphql/resolvers/Pets/types/inputs/CreatePet';

@EntityRepository(Pets)
export default class PetsRepository extends Repository<Pets> {
  async getPets() {
    return this.find();
  }

  async createPet(params: CreatePetParams) {
    return this.save(params);
  }
}
