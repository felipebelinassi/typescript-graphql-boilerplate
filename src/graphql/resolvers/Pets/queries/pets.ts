import { Resolver, Query } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import Pets from '../types/objects/Pets';
import PetsRepository from '../../../../entities/Pets/PetsRepository';

@Resolver()
export default class GetPetsResolver {
  @InjectRepository(PetsRepository)
  private petsRepository!: PetsRepository;

  @Query(() => [Pets], { description: 'List of pets by user' })
  async pets(): Promise<Pets[]> {
    return this.petsRepository.getPets();
  }
}
