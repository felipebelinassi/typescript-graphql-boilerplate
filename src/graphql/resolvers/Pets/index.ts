import pets from './queries/pets';
import createPet from './mutations/createPet';

export default {
  pets,
  createPet,
};
