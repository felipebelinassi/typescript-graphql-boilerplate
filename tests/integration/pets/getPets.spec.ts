import { Connection } from 'typeorm';
import { createTestClient } from 'apollo-server-testing';
import { getPetsQuery } from './__gql__/queries';
import {
  createTestConnection,
  createTestServer,
  clearTestDb,
} from '../../utils';
import Pets from '../../../src/entities/Pets';

let testConnection: Connection;

beforeAll(async () => {
  testConnection = await createTestConnection();
});

beforeEach(async () => {
  await clearTestDb(testConnection);
});

afterAll(async () => {
  await testConnection.close();
});

describe('Get Pets', () => {
  it('should return empty array if no pets are registered', async () => {
    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: getPetsQuery,
    });
    expect(res).toMatchSnapshot('Empty pets list');
  });

  it('should return array with one registered pet', async () => {
    const pet = {
      id: 1,
      name: 'Test dog',
      breed: 'Tester',
      species: 'Canine',
      birthDate: '2021-05-07',
    };

    await testConnection.getRepository(Pets).save(pet);

    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: getPetsQuery,
    });
    expect(res).toMatchSnapshot('Filled pets list');
  });
});
