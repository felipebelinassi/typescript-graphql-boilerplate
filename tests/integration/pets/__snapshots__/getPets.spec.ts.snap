// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Pets should return array with one registered pet: Filled pets list 1`] = `
Object {
  "data": Object {
    "pets": Array [
      Object {
        "birthDate": "2021-05-07",
        "breed": "Tester",
        "id": 1,
        "name": "Test dog",
        "species": "Canine",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Get Pets should return empty array if no pets are registered: Empty pets list 1`] = `
Object {
  "data": Object {
    "pets": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
